### PR TITLE
Fix #7629 - `scheduledForSynchronization` leaks memory when using `@ORM\ChangeTrackingPolicy("DEFERRED_EXPLICIT")

### LINKED ISSUE #7629: `scheduledForSynchronization` leaks memory when using `@ORM\ChangeTrackingPolicy("DEFERRED_EXPLICIT")`
### Bug Report

`scheduledForSynchronization` not erases when no changes on the flush.

|    Q        |   A
|------------ | ------
| BC Break    | yes/no
| Version     | 2.6.3

#### Summary

I use `DEFERRED_EXPLICIT` policy for entities in my app. And app crashes with out of memory error during batch processing. Because `scheduledForSynchronization` keeps references to entities when no changes for commit:

https://github.com/doctrine/orm/blob/434820973cadf2da2d66e7184be370084cc32ca8/lib/Doctrine/ORM/UnitOfWork.php#L359

In this case `UnitOfWork::postCommitCleanup()` not called.

#### How to reproduce

```php

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity()
 * @ORM\ChangeTrackingPolicy("DEFERRED_EXPLICIT")
 */
class Entity
{
}

$entity = $em->find(Entity::class, 123);

$em->persist($entity);
$em->flush();
$em->detach($entity);

var_dump($em->getUnitOfWork()->size()); // int(0)
var_dump($em->getUnitOfWork()->isScheduledForDirtyCheck($entity)); // bool(true)
```

Was expected that `isScheduledForDirtyCheck()` return `false` after commit (flush) in any case.

### PR DESCRIPTION
**Pull request by yethee**
Test to demonstrate unexpected behavior, related to issue #7629

### CODE REVIEW

**Comment by Ocramius** [comment]
So, I think that on `master` we need to take this bit and isolate it to a `private` method to be called also in case of a clean `UnitOfWork`:

https://github.com/doctrine/orm/blob/e8eaf8386de1497863cb400fbc4fddda5878db7e/lib/Doctrine/ORM/UnitOfWork.php#L429-L439

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 1]
```diff
@@ -426,7 +428,11 @@ public function commit()
 
         $this->dispatchPostFlushEvent();
 
-        // Clean up
+        $this->postCommitCleanup();
+    }
+
+    private function postCommitCleanup()
```
Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]:
```suggestion
    private function postCommitCleanup() : void
```

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 1 above)
Comment (yethee) [reply lib/Doctrine/ORM/UnitOfWork.php]: I did not add `void` because other methods without return type-hints.

Comment (Ocramius) [reply lib/Doctrine/ORM/UnitOfWork.php]: `master` has enforced CS fixes for diffs anyway, so if this isn't applied you'll get a red build anyway

Comment (yethee) [reply lib/Doctrine/ORM/UnitOfWork.php]: Fixed

**Review by Ocramius — APPROVED** [review]
Thanks @yethee!

**Pull request merged**